core/recourse_iterator.py

```diff
@@ -25,29 +24,29 @@ class RecourseIterator:
     """Class for iterating recourse."""
     def __init__(self,
                  recourse_method: RecourseMethod,
-                 preprocessor: dp.Preprocessor,
+                 adapter: recourse_adapter.RecourseAdapter,
                  certainty_cutoff: float = None,
                  check_certainty: CertaintyChecker = None):
         """Creates a new recourse iterator.
 
         Args:
             recourse_method: The recourse method to use.
-            preprocessor: The preprocessor.
+            adapter: The adapter.
             certainty_cutoff: If not None, stop iterating early if the model certainty reaches the cutoff.
             check_certainty: If not None, the function used to compute model certainty.
         """
         self.recourse_method = recourse_method
         self.certainty_cutoff = certainty_cutoff
         self.check_certainty = check_certainty
-        self.preprocessor = preprocessor
+        self.adapter = adapter
 
     def iterate_k_recourse_paths(self, poi: pd.Series, max_iterations: int) -> Sequence[pd.DataFrame]:
         """Generates one recourse path for each of the model recourse directions."""
         all_instructions = self.recourse_method.get_all_recourse_instructions(poi)
         # Start the paths from the POI
         cfes = []
```

Comment:
Please use a more descriptive name for cfes and cfe.

https://google.github.io/styleguide/pyguide.html#316-naming